api/app/Repositories/UserRepository.php

```diff
@@ -0,0 +1,19 @@
+<?php
+
+namespace App\Repositories;
+
+use App\Models\User;
+
+class UserRepository
+{
+    /**
+     * get user by email
+     *
+     * @param string $email
+     * @return User
+     */
+    public function getUserByEmail(string $email): User
+    {
+        return User::where("email", $email)->first();
+    }
```

Comment:
getよりfind派です！！  

この下からは余談です  
より抽象的に書くならこんな感じでいけます  

```suggestion
    /**
     * find user by params
     *
     * @param array $params
     * @return User
     */
    public function findUserByParams(array $params): User
    {
        return User::where($params)->first();
    }

    呼び出し
    //emailでwhere
    $this->findUserByParams(['email',$_email]);
    //idでwhere
    $this->findUserByParams(['id',$_id])
```

Reply:
DBから取得する場合はfindですね。勉強になります。

Reply:
配列で受け取れるようにした方が色々なところで使えるから下のほうが良いですね。